packages/Chem/src/rdkit_cell_renderer.js

```diff
@@ -50,26 +41,14 @@ class RDKitCellRenderer extends DG.GridCellRenderer {
     try {
       mol = rdKitModule.get_mol(molString);
       if (mol.is_valid()) {
-        // TODO: maybe split into 2 functions (with scaffold and without)
-        if (molRegenerateCoords) {
-          // "drop" the coordinate information from the molecule
-          let rdkitMolNoCoords = rdKitModule.get_mol(mol.get_smiles());
-          let molBlockString = rdkitMolNoCoords.get_molblock();
-          rdkitMolNoCoords.delete();
-          mol.delete();
-          mol = rdKitModule.get_mol(molBlockString);
-        }
-        if (scaffoldMolString !== "") {
-          // only after dropping, align to a given scaffold
+        if (this._isMolBlock(scaffoldMolString)) {
           let rdkitScaffoldMol = this._fetchMol(scaffoldMolString, "", molRegenerateCoords, false).mol;
-          if (this._molIsInMolBlock(scaffoldMolString, rdkitScaffoldMol)) {
-            substructJson = mol.get_substruct_match(rdkitScaffoldMol);
-            if (substructJson !== '{}') {
-              mol.generate_aligned_coords(rdkitScaffoldMol, true);
-            }
-          }
+          substructJson = mol.generate_aligned_coords(rdkitScaffoldMol, true, true);
+        } else if (molRegenerateCoords) {
+          mol = rdKitModule.get_mol(mol.get_new_coords(true));
```

Comment:
@ptosco should we `mol.delete()` before this to not let memory leak?

Reply:
@dskatov 
Good catch (https://emscripten.org/docs/porting/connecting_cpp_and_javascript/embind.html#memory-management)
I have just submitted another PR to fix this:
https://github.com/datagrok-ai/public/pull/21
Could you please merge it for me? Thanks a lot.